verilog/tools/ls/verilog-language-server.cc

```diff
@@ -0,0 +1,182 @@
+// Copyright 2021-2022 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "verilog/tools/ls/verilog-language-server.h"
+
+// Windows specific implementation of read()
+#ifndef _WIN32
+#include <unistd.h>
+#else
+#include <fcntl.h>
+#include <io.h>
+#include <stdio.h>
+// Windows doesn't have Posix read(), but something called _read
+#define read(fd, buf, size) _read(fd, buf, size)
+#endif
+
+namespace verilog {
+
+void VerilogLanguageServer::JsonRpcDispatcherOutputFormatter(
+    absl::string_view reply) {
+  // Output formatting as header/body chunk as required by LSP spec.
+  std::cout << "Content-Length: " << reply.size() << "\r\n\r\n";
+  std::cout << reply << std::flush;
+}
+
+VerilogLanguageServer::VerilogLanguageServer()
+    : dispatcher_(JsonRpcDispatcherOutputFormatter), buffers_(&dispatcher_) {
+  // All bodies the stream splitter extracts are pushed to the json dispatcher
+  stream_splitter_.SetMessageProcessor(
+      std::bind(&VerilogLanguageServer::MessageParserCallback, this,
+                std::placeholders::_1, std::placeholders::_2));
+
+  // Whenever there is a new parse result ready, use that as an opportunity
```

Comment:
I'd separate out the different initializations in different function calls

  * setting up communication links (stream splitter pushing to dispatcher)
      * This possibly also means that we even might break out the dispatcher_, buffer_ and stream_splitter_ as a separate communication object.
  * wiring up the buffer change listeners
  * wiring up the request handlers